tests/backend_test_base.py

```diff
@@ -284,7 +284,10 @@ def run_tflite(self, tflite_path, feed_dict):
             # tflite sometimes converts from tf but produces an invalid model
             return None, None
 
-    def assert_shapes_correct(self, graph, allow_missing=False, run_checker=True):
+    def assert_shapes_correct(self, graph, allow_missing=False, run_checker=True, check_shape=True):
+        if not check_shape:
```

Comment:
Could you please finish address my comments below and then disable this to see if the validation could pass?

If still fail, could you please share the error details so we can check if it could be corrected?

Reply:
Hi @fatcat-z , thanks for your review. I addressed your previous comments and tried disabling this. The error message is what I saw while trying to run the tests locally.

```
=================================== FAILURES ===================================
__________________ BackendTests.test_conv2d_quantization_axis __________________

self = <test_backend.BackendTests testMethod=test_conv2d_quantization_axis>

    @check_opset_min_version(10, "quantize_and_dequantize")
    def test_conv2d_quantization_axis(self):
        x_shape = [1, 1, 5, 5]
        kernel_shape = _KERNEL3x3
        strides = [1, 1, 1, 1]
        x_val = make_xval(x_shape).transpose(NCHW_TO_NHWC)
        kernel_val = make_xval(_KERNEL3x3)
    
        def func(x):
            f = tf.constant(kernel_val, name="kernel", dtype=tf.float32)
            kernel_dq = quantize_and_dequantize(f, 0, np.prod(kernel_shape))
            conv = tf.nn.conv2d(x, kernel_dq, strides=strides, padding="VALID")
            return tf.identity(conv, name=_TFOUTPUT)
        def graph_validator(g):
            return check_quantization_axis(g, "DequantizeLinear", 0)
    
>       self._run_test_case(func, [_OUTPUT], {_INPUT: x_val}, rtol=1e-05, graph_validator=graph_validator,
                            check_shape=False)

tests/test_backend.py:626: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_backend.py:177: in _run_test_case
    return self.run_test_case(func, feed_dict, [], output_names_with_port, **kwargs)
tests/backend_test_base.py:407: in run_test_case
    self.assert_shapes_correct(g, self.config.allow_missing_shapes, not self.config.skip_onnx_checker,
tests/backend_test_base.py:331: in assert_shapes_correct
    self.assertEqual(onnx_shape, tf2onnx_shape)
E   AssertionError: Lists differ: [1, 1, 3, 3] != [3, 3, 1, 1]
E   
E   First differing element 0:
E   1
E   3
E   
E   - [1, 1, 3, 3]
E   + [3, 3, 1, 1]
```

Essentially the reason why the shapes don't match here is that when the tf model is passed through tf2onnx, the kernel weights are transposed. That's part of the intended behavior of the converter. So this kind of shape mismatch is expected. This is why I think it is fine to ignore the shape check in this case. Let me know if that makes sense.

Reply:
Thanks for your details. Please re-enable your changes and let's submit the final changes.

Reply:
Great, I re-enabled the changes and the branch is now up to date.